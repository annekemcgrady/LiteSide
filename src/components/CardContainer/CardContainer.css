.main {
  border: 1px solid rgb(80, 80, 80);
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(220px, 2fr));
  height: 90vh;
  width: 50%;
  overflow: scroll;
}

.add-fav-notes {
  font-size: 24px;
  color: rgb(134, 133, 133);
  padding-top: 50%;
}

::-webkit-scrollbar {
  width: 5px;
}

::-webkit-scrollbar-track {
  background: #3e3e3e; 
}
 
::-webkit-scrollbar-thumb {
  background: #626262; 
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: #555; 
}