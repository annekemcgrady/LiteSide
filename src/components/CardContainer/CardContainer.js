import React from 'react';
import PropTypes from 'prop-types';
import './CardContainer.css'
import Card from '../Card/Card.js'


const CardContainer = ({ data, handleFavorite, favCount, asideShow }) => {

  const favNote = favCount === 0 && <p className="add-fav-notes">You don't have any favorite cards...</p>;
  
  const mainWidth = asideShow ? '50%' : '80%';

  const mainStyle = favCount === 0 ? {display: 'block', width: mainWidth, transition: '0.5s'} : {display: 'grid', width: mainWidth, transition: '0.5s'};
  
  const Cards = data.map( item => {
    return <Card key={item.id} item={item} handleFavorite={handleFavorite} />
  })


  return (
    
    <main style={mainStyle} className="main">
      { favNote }
      { Cards }
    </main>
  )

  
}

CardContainer.propTypes = {
  data : PropTypes.array,
  handleFavorite : PropTypes.func,
  favCount : PropTypes.number,
  asideShow : PropTypes.bool
}



export default CardContainer;