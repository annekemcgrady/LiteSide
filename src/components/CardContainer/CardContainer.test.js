import React from 'react';
import { shallow, mount } from 'enzyme';
import CardContainer from './CardContainer.js'

describe('CardContainer', ()=> {

it('should match the snapshot with data passed correctly', ()=> {

  const data = [ {
    birth_year: "19BBY",
    created: "2014-12-09T13:50:51.644000Z",
    edited: "2014-12-20T21:17:56.891000Z",
    eye_color: "blue",
    favorite: false,
    films: (5) ["https://swapi.co/api/films/2/", "https://swapi.co/api/films/6/", "https://swapi.co/api/films/3/", "https://swapi.co/api/films/1/", "https://swapi.co/api/films/7/"],
    gender: "male",
    hair_color: "blond",
    height: "172",
    homeworld: "https://swapi.co/api/planets/1/",
    id: "pp-0",
    mass: "77",
    name: "Luke Skywalker",
    skin_color: "fair",
    species: ["https://swapi.co/api/species/1/"],
    starships: (2) ["https://swapi.co/api/starships/12/", "https://swapi.co/api/starships/22/"],
    type: "people",
    url: "https://swapi.co/api/people/1/",
    vehicles: (2) ["https://swapi.co/api/vehicles/14/", "https://swapi.co/api/vehicles/30/"]
  }]

  const wrapper = shallow (<CardContainer
  data={data}
  handleFavorite={jest.fn()} 
  favCount={0}
  />);
  
  expect (wrapper).toMatchSnapshot();

});


});
