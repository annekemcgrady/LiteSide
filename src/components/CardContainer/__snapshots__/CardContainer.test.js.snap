// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with data passed correctly 1`] = `
<main
  className="main"
>
  <Card
    handleFavorite={[MockFunction]}
    item={
      Object {
        "birth_year": "19BBY",
        "created": "2014-12-09T13:50:51.644000Z",
        "edited": "2014-12-20T21:17:56.891000Z",
        "eye_color": "blue",
        "favorite": false,
        "films": undefined,
        "gender": "male",
        "hair_color": "blond",
        "height": "172",
        "homeworld": "https://swapi.co/api/planets/1/",
        "id": "pp-0",
        "mass": "77",
        "name": "Luke Skywalker",
        "skin_color": "fair",
        "species": Array [
          "https://swapi.co/api/species/1/",
        ],
        "starships": undefined,
        "type": "people",
        "url": "https://swapi.co/api/people/1/",
        "vehicles": undefined,
      }
    }
    key="pp-0"
  />
</main>
`;
